lib/common/formats.js

```diff
@@ -347,5 +347,69 @@ module.exports = {
       })
       .value();
     return JSON.stringify(to_ret, null, 2);
+  },
+
+  /**
+   * Creates a flatten JSON from input, capitalizing and then concatenating 
+   * each key with it's pre-key recursively.
+   *
+   * @memberof Formats
+   * @kind member
+   * @example
+   * ```json
+    {
+      "size": {
+        "font": {
+           "small": { "value": "13px" },
+           "medium": { "value": "15px" },
+           "large": { "value": "17px" },
+           "base": { "value": "{size.font.medium.value}" }
+        }
+      }
+    }```
+   * 
+   * Creates:
+   * ```json
+    {
+      "SizeFontSmall": "13px",
+      "SizeFontMedium": "15px",
+      "SizeFontLarge": "17px",
+      "SizeFontBase": "15px"
+    }
+   * ```
+   */
+  'json/flat': function(dictionary) {
+    var evaluateProp = function(property) {
+      // property value is in format '{*}'
+      var isReference = /^\{.*\}/g.test(property);
```

Comment:
there is an open discussion if continue supporting multiple references in the same token value or not (see #118). in theory, the value could be something like `{abc.def} {ghi.jkl}` in that case the code may not work as expected.
@dbanksdesign suggestions here? what do we do?

Reply:
If it's needed to keep support, I just need to know how it should work. Waiting.

Reply:
The style dictionary system already handles the references at this point before getting to the formatter, so you should not be trying to resolve the reference. I think the confusion here is in the example I gave in the issue is referencing the input JSON to a style dictionary, which gets processed by the style dictionary before being passed to a formatter. Take a look at https://amzn.github.io/style-dictionary/#/build_process to see what is happening.  Apologies for any confusion.